### ISSUE TITLE
What URL functions are you using?

### ISSUE THREAD

**Issue by vinkla** [wordplate/framework#78]
WordPlate's goal has always been to enhance the developer experience with WordPress. Making it easier to work together with WordPress with newer standards in PHP while not adding too many custom features.

I've recently looked through which URL functions we've implemented in WordPlate which works similarly to functions in WordPress. With this issue I'm raising the question if it be possible to deprecate our custom URL functions and use WordPress functions instead. 

If you've a minute to spare, please answer these questions about WordPlate's URL functions:

1. Are you using the `asset` function?
1. Are you using the `template_url` function?
1. Are you using the `stylesheet_url` function?
1. When do you use `stylesheet_url` function instead of `template_url`?
1. When do you use `asset`, `template_url` and/or `stylesheet_url` functions instead of [`get_theme_file_uri`](https://developer.wordpress.org/reference/functions/get_theme_file_uri/)?

cc @wordplate/developers

**Comment by fiskhandlarn** [comment]
> 1. Are you using the asset function?

Yes! Directly by itself and like this:

```
function asset_url($url)
{
    return asset('assets/'.$url);
}
```

> 2. Are you using the template_url function?

No.

> 3. Are you using the stylesheet_url function?

No.

> 4. When do you use stylesheet_url function instead of template_url?

Never. But I often use both `stylesheet_path` and `template_path` (so keep them please :)).

> 5. When do you use asset, template_url and/or stylesheet_url functions instead of get_theme_file_uri?

Never have, didn't know about `get_theme_file_uri` until now. Please bear in mind that `get_theme_file_uri` searches the stylesheet directory first and the template directory only if nothing was found in the stylesheet directory. This is not a problem for me though.

**Comment by vinkla** [comment]
> Never have, didn't know about `get_theme_file_uri` until now. Please bear in mind that `get_theme_file_uri` searches the stylesheet directory first and the template directory only if nothing was found in the stylesheet directory. This is not a problem for me though.

I didn't know about it either, this is actually the reasoning behind this issue. Great, that it wont effect you 👍

**Comment by puredazzle** [comment]
1. In a few cases, but I could easily change it to some other function.
2. Nope
3. Nope
4. Se above
5. I use `asset` instead of `get_theme_file_uri` because `asset` got a nicer name and is easier to remember.

**Comment by vinkla** [comment]
> I use `asset` instead of `get_theme_file_uri` because `asset` got a nicer name and is easier to remember.

@puredazzle would you consider using `get_theme_file_uri` if it were to replace all three URL functions?

**Comment by puredazzle** [comment]
> > I use `asset` instead of `get_theme_file_uri` because `asset` got a nicer name and is easier to remember.
> 
> Would you consider using `get_theme_file_uri` if it were to replace all three URL functions?

Yes. I don't use `asset` more then 1-2 times in each project, so this is no biggie for me.

**Comment by fiskhandlarn** [comment]
@vinkla No problem for me neither!

**Comment by vinkla** [comment]
This is now merged with #80.
